### PR TITLE
[WIP] [RFC]  RA-family Renesas EK-RA4M1 and EK-RA6M5 boards preliminary support.

### PR DESCRIPTION
**Pull request by piotrrak**
Hello
There is still many thing, but since other board renesas  ra support was merged, I though  to share to get a feedback sooner than later.

Still lot to be done, not full list of major items:
- [ ] Boards documentation
- [ ] Discuss how to handle odd-ball flash memory in DTS (similar to amtel,sam in some respects)
- [ ] Clocks (CGC driver needs extending for RA6/RA8 series support)
- [ ] pinctrl pins names for functions we probably generate those by some kind of script, writing those by hand is pretty tedious and err-prone - RA8M1/RA8M1 have 224 pins with lot more functions.

### CODE REVIEW

**Comment by piotrrak** [comment]
@soburi Hello, could you please take a look in free time especially at https://github.com/zephyrproject-rtos/zephyr/pull/69315/commits/0dbcab24f8f47f57cb87b2be0f01bb669d32cb3b ?
It is most likely bug fix for `arduino_uno_r4_minima` pin values. It's not used in code base, but it is user facing API.

File: boards/renesas/ek_ra4m3/Kconfig.board [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Copyright (c) 2024 Piotr Rak <piotr.rak@gmail.com>
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_RA4M3_EK
+	bool "Renesas EK-RA4M3 board"
```
Comment (nordicjm) [inline boards/renesas/ek_ra4m3/Kconfig.board:5]: Boards no longer have a choice so
```suggestion
	bool
```

File: boards/renesas/ek_ra4m3/Kconfig.board [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+	depends on SOC_R7FA4M3AF3CFB
```
Comment (nordicjm) [inline boards/renesas/ek_ra4m3/Kconfig.board:6]: And need to select the soc e.g.
```suggestion
	select SOC_R7FA4M3AF3CFB
```

File: boards/renesas/ek_ra4m3/Kconfig.board [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# Copyright (c) 2024 Piotr Rak <piotr.rak@gmail.com>
```
Comment (nordicjm) [inline boards/renesas/ek_ra4m3/Kconfig.board:1]: Needs to be named `Kconfig.<board>` so `Kconfig.ek_ra4m3`

**Review by nordicjm — COMMENTED** [review]
Very quick skim, soc files look fine, changes to board files are needed

File: boards/renesas/ek_ra4m3/Kconfig.board (hunk 1 above)
Comment (piotrrak) [reply boards/renesas/ek_ra4m3/Kconfig.board]: @nordicjm Thanks. Both `Kconfig.board` and `Kconfig.ek_raXX` were present. Simply removed `Kconfig.board` ones.

**Comment by piotrrak** [comment]
Rebased on current collab-hwm after its rebase on 3.6.0

**Comment by soburi** [comment]
> @soburi Hello, could you please take a look in free time especially at [0dbcab2](https://github.com/zephyrproject-rtos/zephyr/commit/0dbcab24f8f47f57cb87b2be0f01bb669d32cb3b) ? It is most likely bug fix for `arduino_uno_r4_minima` pin values. It's not used in code base, but it is user facing API.

@piotrrak 

Thank you for catching my mistakes.
LGTM.

**Comment by piotrrak** [comment]
@soburi  Thank you for looking at this. I've just picked up arduino uno m4 minima and wifi from my mailbox today.  I should be able to test/debug my changes on hardware from now on.

**Comment by soburi** [comment]
@piotrrak 

I have created PR #71174.
It still developing, but I think it may help to update your PR.

**Pull request closed without merging**